File: services/frontend/src/store/modules/posts.js
import axios from 'axios';

const state = {
  posts: null,
  post: null
};

const getters = {
  statePost: state => state.posts,
  statePosts: state => state.post,
};

const actions = {
  async createPost({dispatch}, post){
    await axios.post('posts',post);
    await dispatch('getPosts');
  },

  async getPosts({commit}){
    let {data} = await axios.get('posts');
    commit('setPosts', data);
  },

  async viewPost({commit}, pid){
    let {data} = await axios.get(`post/${pid}`);
    commit('setPost', data);
  },
  async updatePost(post){
    await axios.patch(`post/${post.pid}`, post.form);
  },
  async deletePost(pid){
    await axios.delete(`post/${pid}`);
  }

};

const mutations = {
  setPosts(state, posts){
    state.posts = posts;
  },
  setPost(state, post){
    state.post = post
  }
};

export default {
  state,
  getters,
  actions,
  mutations
};

File: services/backend/src/crud/posts.py
from os import stat
from fastapi import HTTPException, APIRouter, status, Depends
from psycopg2 import connect
from src.database.models import Posts,get_db
from src.schemas.posts import Post, ShowPost
from sqlalchemy.orm import Session

router = APIRouter(
  tags=['posts']
)

connect_db = get_db

async def get_posts(db):
    return await db.query(Posts).all()

async def get_post(pid: int, db: Session=Depends(connect_db)):
    return await db.query(Posts).filter(Posts.pid == pid).all()

# @router.post('/newpost')
async def create_post(request: Post, current_user, db: Session=Depends(connect_db)):
    data = Posts(ptitle= request.ptitle,
                  pcontent=request.pcontent,
                  author = current_user
                  )
    db.add(data)
    db.commit()
    db.refresh(data)
    return "posting success"

async def update_post(pid: int, request, current_user, db):
    try:
        post = db.query(Posts).filter(Posts.pid == pid).first()
    except:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"post {pid} not found"
            )
    if post['author'] == current_user.uname:
        data = Posts(ptitle = request.ptitle, pcontent = request.pcontent)
        db.update(data)
        db.commit()
        db.refresh(data)
        return await db.query(Posts).filter(Posts.pid == pid).first()

async def delete_post(pid: int, current_user, db):
    try:
        post = db.query(Posts).filter(Posts.pid == pid).first()
    except:
        raise HTTPException(
          status_code=status.HTTP_404_NOT_FOUND,
          detail=f"post {pid} not found"
        )
    if post['author'] == current_user.uname:
        post.delete(synchronize_session=False)
        db.commit()
        return f"User '{pid}' deleted"
